rct229/rulesets/ashrae9012019/section12/section12rule1.py

```diff
@@ -11,35 +17,95 @@ class Section12Rule1(RuleDefinitionListIndexedBase):
             rmds_used=produce_ruleset_model_description(
-                USER=True, BASELINE_0=True, PROPOSED=False
+                USER=False, BASELINE_0=True, PROPOSED=True
             ),
-            each_rule=Section12Rule1.BuildingRule(),
-            index_rmd=USER,
+            each_rule=Section12Rule1.RMDRule(),
+            index_rmd=BASELINE_0,
             id="12-1",
             description=(
-                "Number of spaces modeled in User RMD and Baseline RMD are the same"
+                "Receptacle and process power shall be modeled as identical to the proposed design"
             ),
             ruleset_section_title="Receptacle",
             standard_section="Section Table G3.1-12 Modeling Requirements for the Proposed design",
             is_primary_rule=True,
-            rmd_context="ruleset_model_descriptions/0/buildings",
+            list_path="ruleset_model_descriptions[0]",
+            data_items={"compliance_path": (BASELINE_0, "compliance_path")},
         )
 
-    class BuildingRule(RuleDefinitionBase):
+    class RMDRule(RuleDefinitionBase):
         def __init__(self):
-            super(Section12Rule1.BuildingRule, self).__init__(
+            super(Section12Rule1.RMDRule, self).__init__(
                 rmds_used=produce_ruleset_model_description(
-                    USER=True, BASELINE_0=True, PROPOSED=False
+                    USER=False, BASELINE_0=True, PROPOSED=True
                 ),
+                manual_check_required_msg=MANUAL_CHECK_REQUIRED_MSG,
             )
 
         def get_calc_vals(self, context, data=None):
-            user_spaces = find_all("$..spaces[*]", context.USER)
-            baseline_spaces = find_all("$..spaces[*]", context.BASELINE_0)
+            rmd_b = context.BASELINE_0
+            rmd_p = context.PROPOSED
+            unexpected_misc_equipment_power = []
+            reduced_misc_equipment_power = []
+            misc_equipment_list_b = find_all(
+                "$.buildings[*].building_segments[*].zones[*].spaces[*].miscellaneous_equipment[*]",
+                rmd_b,
+            )
+            misc_equipment_list_p = find_all(
+                "$.buildings[*].building_segments[*].zones[*].spaces[*].miscellaneous_equipment[*]",
+                rmd_p,
+            )
+            # This assumes that the miscellaneous_equipment all match
+            matched_misc_equipment_list_p = match_lists_by_id(
+                misc_equipment_list_b, misc_equipment_list_p
+            )
+            proposed_baseline_misc_equipment_pairs = zip(
+                misc_equipment_list_b, matched_misc_equipment_list_p
+            )
+
+            for (
+                misc_equipment_b,
+                misc_equipment_p,
+            ) in proposed_baseline_misc_equipment_pairs:
+                misc_equipment_power_b = misc_equipment_b.get("power", ZERO.POWER)
+                misc_equipment_power_p = misc_equipment_p.get("power", ZERO.POWER)
```

Comment:
@JacksonJ-KC After reading this code, I wonder if we ever considered what means equal between `misc_equipment_b` and `misc_equipment_p`. Have we ever discussed whether the rule's scope is limited in `power` or the object? What if the `misc_equipment_b` has a different energy type but same power as `misc_equipment_p`, do they consider equal in this rule?

Reply:
Good point Weili, I don't think this came up during discussion in the RDS meetings. I think the energy type should also need to match for the rule to pass.